File: snippets/node/20-search/01-find-by-index.mjs
// import the shared client:  import { connect } from "elo-playground";
// topic:    Find objects by index field
// category: Search
// id:       search.find-by-index

import { connect } from "elo-playground";

// --- local ELO test box (override with ELOPG_* env vars or a .env) ---
const ELO_BASE_URL = "http://localhost:9090/ix-Repository1";
const ELO_USER = "Administrator";
const ELO_PASS = "elo";

const elo = await connect({ baseUrl: ELO_BASE_URL, user: ELO_USER, password: ELO_PASS });

const rows = await elo.findAll("findFirstSords", "findNextSords", "sords", {
  findInfo: { findByIndex: { name: "Invoice *", exactName: false } },
  max: 50,
  sordZ: { bset: "449304431574384639" },
});

console.log(`${rows.length} matches`);
for (const s of rows) console.log(`  ${s.name}  (mask: ${s.maskName})`);

File: snippets/python/20-search/01-find-by-index.py
# import the shared client:  from elo_playground import connect
# topic:    Find objects by index field
# category: Search
# id:       search.find-by-index

import os
from elo_playground import connect

# --- local ELO test box (override with ELOPG_* env vars or a .env) ---
ELO_BASE_URL = os.getenv("ELOPG_ELO_BASE_URL", "http://localhost:9090/ix-Repository1")
ELO_USER = os.getenv("ELOPG_ELO_USER", "Administrator")
ELO_PASS = os.getenv("ELOPG_ELO_PASSWORD", "")

elo = connect(base_url=ELO_BASE_URL, user=ELO_USER, password=ELO_PASS)

rows = elo.find_all(
    "findFirstSords", "findNextSords", "sords",
    {
        "findInfo": {"findByIndex": {
            "name": "Invoice *",        # wildcard match on the object name
            "exactName": False,
        }},
        "max": 50,
        "sordZ": {"bset": "449304431574384639"},
    },
)

print(f"{len(rows)} matches")
for s in rows:
    print(f"  {s['name']}  (mask: {s.get('maskName')})")
